src/main/java/org/zone/region/flag/meta/join/visibility/flag/ZoneVisibilityFlagType.java

```diff
@@ -35,18 +35,19 @@
     @Override
     public @NotNull ZoneVisibilityFlag load(@NotNull ConfigurationNode node)
             throws IOException {
+        String name = node.node("ZoneVisibility").getString();
+        if (name == null) {
+            throw new IOException("Unknown Visibility");
+        }
         try {
-            String name = node.node("ZoneVisibility").getString();
-            if (name != null) {
-                ZoneVisibility zoneVisibilityName = ZoneVisibility.valueOf(name);
-                ZoneVisibilityFlag zoneVisibilityFlag = new ZoneVisibilityFlag();
-                zoneVisibilityFlag.setZoneVisibility(zoneVisibilityName);
-                return zoneVisibilityFlag;
-            }
+            ZoneVisibility zoneVisibilityName = ZoneVisibility.valueOf(name);
```

Comment:
Try something like this

```java
ZoneVisibility visibility;
try{
visibility = ZoneVisibility.valueOf(name);
}catch(IllegalArgumentException e){
    throw new IOException("Invalid name");
}
//rest of your code

```

This way it takes all the try and catch code out of the try and catch making it easier to handle and read

Reply:
K

Reply:
cant

Reply:
done